src/transformers/typescript.ts

```diff
@@ -167,6 +293,110 @@ const transformer: Transformer<Options.Typescript> = ({
     }
   }
 
+  if (sourceMapChain) {
+    const fname = markup ? `${filename}.transpiled.js` : `${filename}.js`;
+
+    sourceMapChain.content[fname] = transpiledCode;
+    sourceMapChain.sourcemaps[fname] = JSON.parse(sourceMapText);
+  }
+
+  return { transpiledCode, diagnostics };
+}
+
+export function loadTsconfig(
+  compilerOptionsJSON: any,
+  filename: string,
+  tsOptions: Options.Typescript,
+) {
+  if (typeof tsOptions.tsconfigFile === 'boolean') {
+    return { errors: [], options: compilerOptionsJSON };
+  }
+
+  let basePath = process.cwd();
+
+  const fileDirectory = (tsOptions.tsconfigDirectory ||
+    dirname(filename)) as string;
+
+  let tsconfigFile =
+    tsOptions.tsconfigFile ||
+    ts.findConfigFile(fileDirectory, ts.sys.fileExists);
+
+  tsconfigFile = isAbsolute(tsconfigFile)
+    ? tsconfigFile
+    : join(basePath, tsconfigFile);
+
+  basePath = dirname(tsconfigFile);
+
+  const { error, config } = ts.readConfigFile(tsconfigFile, ts.sys.readFile);
+
+  if (error) {
+    throw new Error(formatDiagnostics(error, basePath));
+  }
+
+  // Do this so TS will not search for initial files which might take a while
+  config.include = [];
+
+  let { errors, options } = ts.parseJsonConfigFileContent(
+    config,
+    ts.sys,
+    basePath,
+    compilerOptionsJSON,
+    tsconfigFile,
+  );
+
+  // Filter out "no files found error"
+  errors = errors.filter((d) => d.code !== 18003);
+
+  return { errors, options };
+}
+
+const transformer: Transformer<Options.Typescript> = async ({
+  content,
+  filename = 'source.svelte',
+  markup,
+  options = {},
+  attributes,
+}) => {
+  const basePath = process.cwd();
+  const compilerOptions = getCompilerOptions({ filename, options, basePath });
+
+  const sourceMapChain = createSourceMapChain({
+    filename,
+    content,
+    compilerOptions,
+  });
+
+  const injectedCode = injectVarsToCode({
+    content,
+    markup,
+    filename,
+    attributes,
+    sourceMapChain,
+  });
+
+  const { transpiledCode, diagnostics } = transpileTs({
+    code: injectedCode,
+    markup,
+    filename,
+    basePath,
+    options,
+    compilerOptions,
+    sourceMapChain,
+  });
+
+  const code = stripInjectedCode({
+    transpiledCode,
+    markup,
+    filename,
+    sourceMapChain,
+  });
+
```

Comment:
This fixes the sorcery error that happens on empty code/maps. A test case should be added to handle this.

```suggestion

if (!code) {
  return { code, diagnostics };
}

```